src/lp_pricer/lp_pricer.cairo

```diff
@@ -0,0 +1,391 @@
+            let (token_a_id, token_b_id) = fetch_tokens_symbols(token_a_address, token_b_address);
+
+            // [Check] Assert that both pool assets are supported by Pragma
+            let oracle = self.oracle.read();
+            assert(currency_is_supported(oracle, token_a_id), errors::UNSUPPORTED_CURRENCY);
+            assert(currency_is_supported(oracle, token_b_id), errors::UNSUPPORTED_CURRENCY);
+
+            // [Effect] Add the pool to the storage
+            let pool = Pool {
+                id: pool_dispatcher.symbol(),
+                token_a: TokenTrait::new(token_a_id, token_a_address),
+                token_b: TokenTrait::new(token_b_id, token_b_address),
+                total_supply: pool_dispatcher.total_supply(),
+                dispatcher: pool_dispatcher,
+            };
+            self.supported_pools.write(pool_address, pool);
+
+            // [Interaction] Pool registered event
+            self
+                .emit(
+                    Event::RegisteredPool(
+                        RegisteredPool {
+                            pool_symbol: pool.id, pool_address: pool.dispatcher.contract_address,
+                        }
+                    )
+                );
+        }
+
+        /// Removes a pool from the supported list.
+        /// 
+        /// Can only be called by the admin.
+        fn remove_pool(ref self: ContractState, pool_address: ContractAddress) {
+            // [Check] Only admin
+            assert_only_admin();
+            // [Check] Pool is registered
+            assert(self.is_supported_pool(pool_address), errors::UNSUPPORTED_POOL);
+
+            // [Effect] Remove the Pool contract from the storage
+            let pool = self.supported_pools.read(pool_address);
+            self.supported_pools.write(pool_address, PoolTrait::zero());
+
+            // [Interaction] Pool unregistered event
+            self
+                .emit(
+                    Event::RemovedPool(
+                        RemovedPool { pool_symbol: pool.id, pool_address: pool_address, }
+                    )
+                );
+        }
+
+        /// Retrieves information about a Pool, i.e its name, the symbol, the address, the
+        /// decimals and the total supply.
+        fn get_pool_info(self: @ContractState, pool_address: ContractAddress) -> PoolInfo {
+            // [Check] Pool is registered
+            assert(self.is_supported_pool(pool_address), errors::UNSUPPORTED_POOL);
+
+            // [Effect] Retrieve the pool
+            let pool = self.supported_pools.read(pool_address);
+
+            // [Interaction] Return the pool informations
+            PoolInfo {
+                address: pool.dispatcher.contract_address,
+                name: pool.dispatcher.name(),
+                symbol: pool.id,
+                decimals: pool.dispatcher.decimals(),
+                total_supply: pool.total_supply,
```

Comment:
I think we might need to fetch the total supply from the pool, instead of the storage to avoid stale data

Reply:
The total supply can never change, no? We already fetch it when adding it to the pool, not sure if that's enough